[CS] Add a missing null check in repairFailures

hamishknight opened the pull request:
If we fail to simplify the locator, such as in the case where we have synthesized a missing call argument, bail without trying to add a missing call fix.

Resolves SR-12964.
Resolves rdar://64168162.

hamishknight commented:
@swift-ci please smoke test

xedin commented on lib/Sema/CSSimplify.cpp:
```diff
@@ -3048,7 +3048,7 @@ bool ConstraintSystem::repairFailures(
     if (fnType->getNumParams() > 0) {
       auto anchor = simplifyLocatorToAnchor(getConstraintLocator(locator));
-      if (!anchor.is<Expr *>())
+      if (!anchor || !anchor.is<Expr *>())
```
Ah, you can use `!isExpr(anchor)` here instead, that would do a null check internally. Looks like I missed this place while doing switch to `ASTNode` :(

hamishknight replied:
Ah yeah nice, in fact let's use `getAsExpr` exclusively here

hamishknight committed:
[CS] Add a missing null check in repairFailures

If we fail to simplify the locator, such as in the
case where we have synthesized a missing call
argument, bail without trying to add a missing
call fix.

Resolves SR-12964.
Resolves rdar://64168162.

hamishknight commented:
@swift-ci please smoke test

hamishknight commented:
@swift-ci please smoke test

hamishknight commented:
@swift-ci please smoke test macOS

hamishknight commented:
@swift-ci please smoke test macOS
